### PR TITLE
Remove noqa: G004 warnings

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132917
* #132888

Remove logging messages with f-strings (G004), https://docs.astral.sh/ruff/rules/logging-f-string/

cc @XilunWu @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Review by c-p-i-o — APPROVED** [review]

File: torch/distributed/pipelining/schedules.py [hunk 1]
```diff
@@ -1845,6 +1839,8 @@ def need_bubble(stage, op, microbatch, num_stages_global, seen_ops):
 
         if total_bubbles_added > 0:
             logger.warning(
-                f"Non zero bubbles added: {total_bubbles_added=} {bubbles_added=}"  # noqa: G004
+                "Non zero bubbles added: total_bubbles_added=%s bubbles_added=%s",
```
Comment (Skylion007) [inline torch/distributed/pipelining/schedules.py:1842]: nit should be `%d` ?

File: torch/distributed/pipelining/schedules.py [hunk 2]
```diff
@@ -504,7 +504,7 @@ def _batch_p2p(p2p_ops: List[dist.P2POp], desc: Optional[str] = None):
     if len(p2p_ops) == 0:
         return None
     desc_str = f"{desc}, " if desc else ""
```
Comment (Skylion007) [inline torch/distributed/pipelining/schedules.py:506]: this is only used for logging, can it be a LazyStr?

**Review by fduwjj — APPROVED** [review]

**Review by fegin — APPROVED** [review]

**Pull request closed without merging**